### DISCUSSION TITLE
Built-In Environment Variables are no longer working

### DISCUSSION THREAD

**Discussion by dustin-graham** [codemagic-ci-cd/codemagic-docs#2345] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

in a pipeline, try to run a command which utilizes a built-in environment variable:
```
widgetbook publish --api-key $WIDGETBOOK_CLI_KEY --base-branch $CM_PULL_REQUEST_DEST
```

The value for `CM_PULL_REQUEST_DEST` is empty at build time and causes this request to fail. This has been working for months and is now failing without any changes to the pipeline.

I ssh'd to the build box and found that most of the built-in variables return empty right now.

### Expected results

`CM_PULL_REQUEST_DEST` should return the value of the destination pull request branch.

### Actual results

`CM_PULL_REQUEST_DEST` returns empty

note: we have billing enabled but the chat support bubble is missing now. Not sure what has happened there.

### Build id (optional)

65bb02bb13c644e6d10c6a16

**Comment by dtrdic** [comment]
hello @dustin-graham how do you use this variable, are you sure that the build is triggered by pull request? i just tried  to create a pull request to my `main` branch and i can see that the  CM_PULL_REQUEST_DEST variable is displaying the correct value
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/86314670-152a-4f63-9ddc-de081994550a)

**Comment by dtrdic** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
